### PR TITLE
[themes] Allow styling of SQL/python/expression/HTML code editors

### PR DESCRIPTION
**Pull request by nirvn**
## Description
Themes didn't play well with our qsintilla-based editors. This ends now:
image

image

image

@nyalldawson , I'd love a review on this one.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/gui/qgscodeeditor.cpp [hunk 1]
```diff
@@ -90,16 +92,35 @@ void QgsCodeEditor::keyPressEvent( QKeyEvent *event )
...
+    for ( const auto &key : ini.allKeys() )
+    {
+      colors.insert( key, QgsSymbolLayerUtils::decodeColor( ini.value( key ).toString() ) );
+    }
+  }
+
   setUtf8( true );
   setCaretLineVisible( true );
-  setCaretLineBackgroundColor( QColor( 252, 243, 237 ) );
+  setCaretLineBackgroundColor( colors.value( QStringLiteral( "caretLineColor" ), QColor( "#fcf3ed" ) ) );
```
Comment (nyalldawson) [inline src/gui/qgscodeeditor.cpp]: Can you change the string literals for the default color to the numeric QColor constructor instead? It's more efficient then the string parser.

File: src/gui/qgscodeeditor.cpp [hunk 2]
```diff
@@ -90,16 +92,35 @@ void QgsCodeEditor::keyPressEvent( QKeyEvent *event )
...
+  setSelectionBackgroundColor( colors.value( QStringLiteral( "selectionBackgroundColor" ), QColor( "#d7d7d7" ) ) );
 
   setBraceMatching( QsciScintilla::SloppyBraceMatch );
-  setMatchedBraceBackgroundColor( QColor( 183, 249, 7 ) );
+  setMatchedBraceBackgroundColor( colors.value( QStringLiteral( "matchedBraceColor" ), QColor( "#b7f907" ) ) );
   // whether margin will be shown
   setMarginVisible( mMargin );
+  setMarginsForegroundColor( colors.value( QStringLiteral( "marginForegroundColor" ), QColor( "#3e3ee3" ) ) );
+  setMarginsBackgroundColor( colors.value( QStringLiteral( "marginBackgroundColor" ), QColor( "#f9f9f9" ) ) );
+  setIndentationGuidesForegroundColor( colors.value( QStringLiteral( "marginForegroundColor" ), QColor( "#3e3ee3" ) ) );
```
Comment (nyalldawson) [inline src/gui/qgscodeeditor.cpp]: Where do these default colors come from?

**Review by nyalldawson — COMMENTED** [review]
Looks great to me! 3.6 is truely the era of the dark theme!

**Comment by nyalldawson** [comment]
One thing that stands out now -- those bad default colors under the expression editor

**Comment by nirvn** [comment]
@nyalldawson , hex values were all converted to int, int, int.

I've taken your other remarks as a queue to update the default colors. I've based the updated colors on the work we had done a while ago on coming up with a new set of default colors for the console.

**Pull request merged**